yjit/src/asm/arm64/inst/data_processing_immediate.rs

```diff
@@ -0,0 +1,198 @@
+use super::{
+    super::opnd::*,
+    family::Family,
+    sf::Sf
+};
+
+/// The operation being performed by this instruction.
+enum Op {
+    Add = 0b0,
+    Sub = 0b1
+}
+
+// Whether or not to update the flags when this instruction is performed.
+enum S {
+    LeaveFlags = 0b0,
+    UpdateFlags = 0b1
+}
+
+/// How much to shift the immediate by.
+enum Shift {
+    LSL0 = 0b0,
+    LSL12 = 0b1
```

Comment:
Asking just because I'm trying to understand LSL12. It doesn't really make sense for the shift amount to be 12 bits (that's a huge number), so this must be shifting a 12-bit immediate, but by how many bits? 😮

Reply:
I _think_ the answer is that it's shifted by 12 bits. See here: https://developer.arm.com/documentation/ddi0596/2021-12/Base-Instructions/ADD--immediate---Add--immediate--?lang=en.